d2cli/main.go

```diff
@@ -329,10 +338,22 @@ func Run(ctx context.Context, ms *xmain.State) (err error) {
 		return w.run()
 	}
 
+	var singleBoard bool
+	switch {
+	case boardFlag == nil:
+	case *boardFlag == "":
+		boardFlag = nil
+	default:
+		*boardFlag = strings.Replace(*boardFlag, ".", "/", -1)
```

Comment:
Should do some validation on the input first, even though flag instructions are there. I think just checking that `root` is prefix and `/` doesn't exist should be fine

Reply:
why do we need to always prefix the argument with root, just to trim it out? It feels more like boilerplate than useful info and you have to type it each use.

- `--target root` is the same as not providing the target == `-target ''` so it doesn't seem like a useful option
- `--target root.x` can just be `--target x` (with future object targeting, if there is an object also named `x`, then `--target layers.x` would be necessary (in future object has priority over layer))

Noting that these are object paths and not file paths, but `x -> "y / b"` is valid so a layer name could have a `/`.
I think for validation it should just see if it can find the object.

Reply:
oh `--target root` is when you don't want any scenarios

Reply:
There is a difference:

* `--target ''` would render entire diagram, with all layers, scenarios and steps
* `--target root` would render just root and no children

To allow writing to stdout (which is my main motivation for this feature), this distinction is important.

If `/` is a valid character in layer names, wouldn't it also break this existing code?

```go
// boardPath comes in the form of "x/layers/z/scenarios/a"
// or "layers/z/scenarios/a"
// or "x/z/a"
func (d *Diagram) GetBoard(boardPath string) *Diagram {
	path := strings.Split(boardPath, string(os.PathSeparator))
	if len(path) == 0 || len(boardPath) == 0 {
		return d
	}
```

If so, it seems like a bug.

Reply:
`d2 index.d2` would render entire diagram with all layers scenarios steps
`d2 --target '' index.d2` would render just root and no children

that code does seem like it has a bug for layer names with `/`

Reply:
Default value is an empty string (because `nil` is not possible), so `d2 --target `` index.d2` is the same as `d2 index.d2`. To render just root and no children, `--target root` is required.

```bash
❯ go run main.go --target root index.d2
success: successfully compiled index.d2 to index.svg in 69.461959ms

❯ go run main.go --target '' index.d2
success: successfully compiled index.d2 to index/x.svg in 70.542209ms
success: successfully compiled index.d2 to index/index.svg in 69.001958ms

❯ go run main.go index.d2   
success: successfully compiled index.d2 to index/x.svg in 71.247833ms
success: successfully compiled index.d2 to index/index.svg in 69.866416ms
```

Reply:
So I would propose:
 - only check if `--target` starts with `root`
 - all characters are allowed

The bug is in an already existing code and I don't know enough about D2 internals to fix it, so I think it's best to open a new issue and fix it there.

BTW is changing all dots to `/` safe?

Reply:
hmm maybe `*` should be the default value since it will target everything by default. Then it is possible to detect that the target flag was provided with `d2 --target ' index.d2`.

Reply:
I think these will need to be properly parsed, just doing string manipulations isn't exactly safe, but we can make this change when fixing #1727 .
<img width="981" alt="Screenshot 2023-11-13 at 12 12 26 PM" src="https://github.com/terrastruct/d2/assets/85081687/354f47fc-61de-4e88-95dd-5a798955b7ab">

Reply:
`*` would add a lot of confusion, since this, being a default value, would be visible in `--help` output. It's not obvious that you need to override the default value with nothing (empty string) to render root with no children. And also if `*` renders everything, I would expect `x.*` to render `x` and all of its children, which is not the case.

Reply:
IMO `--target root` and `--target root.x` is easier to understand and use, especially since you can already see it in some places, for example PDF output.

```d2
a -> b

layers: {
  x / y: {
    c -> d
  }
}
```

<img width="922" alt="Screenshot 2023-11-13 at 21 29 10" src="https://github.com/terrastruct/d2/assets/13437951/90ece32b-9b41-49ea-b0e8-69d031d3307d">

<img width="922" alt="Screenshot 2023-11-13 at 21 29 02" src="https://github.com/terrastruct/d2/assets/13437951/948df83a-7aa0-4b47-a5d6-4acbf64d96bd">

Reply:
> `*` would add a lot of confusion, since this, being a default value, would be visible in `--help` output. It's not obvious that you need to override the default value with nothing (empty string) to render root with no children. And also if `*` renders everything, I would expect `x.*` to render `x` and all of its children, which is not the case.

I think `x.*` to render just a specific scenario and its steps sounds like a good feature to eventually have. We can update the description to explain that `'' or root` can be used to target only the root.

Reply:
note that the `root` pdf label we be able to be renamed #1644

Reply:
`*` it shall be. I don't want to introduce half-baked feature, so I will add support for `x.*`. 

> `''` or `root` can be used to target only the root

I don't think this is a good idea since `root` is a valid layer name and there could be a conflict. With good documentation `''` should be enough.

I will ping you when changes are ready.

Reply:
@gavin-ts 

I updated the PR. It even works with this diagram now:

```d2
layers: {
  a: {
    layers: {
      "x / y . z": {
        mad
      }
    }
  }
}
```

```bash
❯ go run main.go --target a."\"x / y . z\"" index.d2
success: successfully compiled index.d2 to index.svg in 27.507ms
```

And asterisk allows rendering specific board with all of its children:

```d2
x
layers: {
  a: {
    layers: {
      b: {b}
      c: {c}
    }
  }
}
```

```bash
❯ go run main.go --target=a.* index.d2
success: successfully compiled index.d2 to index/a/b.svg in 60.910375ms
success: successfully compiled index.d2 to index/a/c.svg in 59.784417ms
```